### ISSUE TITLE
Teuchos: CommandLineProcessor: depending on position --help may show incorrect default values

### ISSUE THREAD

**Issue by aprokop** [trilinos/Trilinos#585]
@trilinos/teuchos @bartlettroscoe 

```
$ ./MueLu_ReuseSequence.exe --xml=reuse_ml.xml  --help    --nx=400 --ny=400 --dim=2
...
--dim                  int           space dimension (default: --dim=3)

$ ./MueLu_ReuseSequence.exe --xml=reuse_ml.xml     --nx=400 --ny=400 --dim=2 --help
...
--dim                  int           space dimension (default: --dim=2)
```

**Comment by bartlettroscoe** [comment]
Should be easy to fix.  How urgent is this?

**Comment by aprokop** [comment]
Not urgent at all.

**Comment by bartlettroscoe** [comment]
Now that PR #1324 is merged, this should be addressed.  Moving to in reveiw.

@aprokop, please close when you can confirm this is fixed (but since you create the PR #1324, then my guess is it will).
